controllers/machine_controller_node_labels_test.go

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha4"
+)
+
+func TestReconcileInterruptibleNodeLabel(t *testing.T) {
+	g := NewWithT(t)
+
+	g.Expect(clusterv1.AddToScheme(scheme.Scheme)).To(Succeed())
+
+	r := &MachineReconciler{
+		Client:   fake.NewFakeClientWithScheme(scheme.Scheme),
```

Comment:
Any chance you could convert this to a test that uses the envtest available for the suite, instead of the fake client? You'd essentially create a Machine, InfraMachine, Node with all the right values filled in, then do an Eventually block to check to see if the Node's interruptible label is set.